File: src/main/java/com/examportal/entities/StudentQuery.java
package com.examportal.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name = "student_queries")
@NoArgsConstructor
@Getter
@Setter
@ToString(callSuper = true)
public class StudentQuery extends BaseEntity {
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private User student;
    
    @Column(length = 50, nullable = false)
    private String username;
    
    @Column(length = 100, nullable = false)
    private String title;
    
    @Column(length = 50, nullable = false)
    private String subject;
    
    @Column(columnDefinition = "TEXT", nullable = false)
    private String query;
    
    @Column(columnDefinition = "TEXT")
    private String response;
    
    @Column(length = 20, nullable = false)
    private String status = "PENDING";
    
    @Column(name = "responded_at")
    private java.time.LocalDateTime respondedAt;

    public StudentQuery(User student, String title, String subject, String query) {
        this.student = student;
        this.username = student.getUsername();
        this.title = title;
        this.subject = subject;
        this.query = query;
        this.status = "PENDING";
    }
}

File: Dotnet3/ExamPortal/Models/StudentQuery.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ExamPortal.Models
{
    public class StudentQuery : BaseEntity
    {
        [Required]
        public long StudentId { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string Username { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string Title { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(50)]
        public string Subject { get; set; } = string.Empty;
        
        [Required]
        public string Query { get; set; } = string.Empty;
        
        public string? Response { get; set; }
        
        [Required]
        [MaxLength(20)]
        public string Status { get; set; } = "PENDING";
        
        public DateTime? RespondedAt { get; set; }
        
        // Navigation properties
        [ForeignKey("StudentId")]
        public virtual User Student { get; set; } = null!;
    }
}
